.github/workflows/nightly.yml

```diff
@@ -0,0 +1,65 @@
+name: Maven Build for Snapshot
+
+# Performs a Maven release using JDK 11 on a schedule
+on:
+  release:
+    types: [ published ]
+
+  schedule:
+    - cron:  '20 20 * * *'
+
+jobs:
+  publish:
+    name: Maven Artifact Publisher (JDK ${{ matrix.java }} with ${{ matrix.build_property }})
+    runs-on: ubuntu-latest
+    env:
+      SKIP_JAR_DEPLOYMENT: ${{ secrets.SKIP_JAR_DEPLOYMENT }}
+    strategy:
+      matrix:
+        java: [ 11 ]
+        build_property: [ "kakadu.version", "ignored.version" ]
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2
+      - name: Install JDK ${{ matrix.java }}
+        uses: actions/setup-java@d202f5dbf7256730fb690ec59f6381650114feb2 # v1
+        with:
+          java-version: ${{ matrix.java }}
+          cache: maven
+      - name: Set Jar deployment config
+        if: env.SKIP_JAR_DEPLOYMENT == null
+        run: echo "SKIP_JAR_DEPLOYMENT=false" >> $GITHUB_ENV
+      - name: Optionally, login to Docker repository
+        uses: docker/login-action@f3364599c6aa293cdc2b8391b1b56d0c30e45c8a # v1.8.0
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+        if: env.DOCKER_USERNAME != null
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Optionally, set artifact qualifier (for private builds that include Kakadu)
+        if: matrix.build_property == 'kakadu.version'
+        run: echo "ARTIFACT_QUALIFIER=-ucla" >> $GITHUB_ENV
+      - name: Install Kakadu SSH key
+        uses: webfactory/ssh-agent@ee29fafb6aa450493bac9136b346e51ea60a8b5e # v0.4.1
+        with:
+          ssh-private-key: ${{ secrets.KAKADU_PRIVATE_SSH_KEY }}
+      - name: Release with Maven
+        uses: samuelmeuli/action-maven-publish@201a45a3f311b2ee888f252ba9f4194257545709 # v1.4.0
+        with:
+          gpg_private_key: ${{ secrets.BUILD_KEY }}
+          gpg_passphrase: ${{ secrets.BUILD_PASSPHRASE }}
+          nexus_username: ${{ secrets.SONATYPE_USERNAME }} # These are placeholders; we're publishing a Docker image
+          nexus_password: ${{ secrets.SONATYPE_PASSWORD }}
+          maven_args: >
+            -Drevision=nightly
+            -Dcantaloupe.version=dev
+            -Dcantaloupe.commit.ref=latest
```

Comment:
So here is the question now that we've condensed the nightly builds in build.yml and release.yml into one nightly build... do we dilute the usefulness of a nightly build to catch OS changes if we're not using the same code that we use in the production build (the commit hash of the tagged version we use). I guess, no, since the codebase is different from the OS packages? Might we get into a situation where the latest code needs different OS dependencies? Is it okay to deal with it then (and how do we plant the seed for us to remember this?)

How about for our purposes? We want an image to test the patches we want to use before we put them into production. Is this valuable if we're testing with a different version of the code than the one we want to put into production?

The nightly build serves a couple of purposes:
* Run a nightly check against our Docker image's pinned dependencies/packages to see if any become obsolete
* Give us a snapshot of the latest version of Cantaloupe with the latest and greatest features and bug fixes
* Give us a way to test patches before we move them into production

Did I miss any other purposes? Do we have a conflict of purposes by condensing the nightly builds into one?

Reply:
This may tie in with your original idea (I think on another related ticket) to do a pre-release version, which could use the same nightly build here but use a GHA matrix build.

Reply:
> how do we plant the seed for us to remember this?

I suppose we could run both builds ("release" and "snapshot") on PRs, and if one of them fails due to OS dep mismatch, address the issue at that point. I did not realize that the scheduling of `build.yml` was intended to catch OS changes in the "release" build, since the workflow is titled "Maven PR Build".

> How about for our purposes?

I have been working off the assumption that our nightly build wouldn't use the patch for long, in hopes that it would be fixed upstream "soon enough". For me, the immediate need it satisfies is providing, for the cantaloupe-auth-delegate integration tests, a version of Cantaloupe without (AFAIK) a buggy auth implementation. At the point that we consider the patched image safe to use in production, I would hope we've already submitted the patch in an upstream PR. I would like to stop using the patch in our nightly builds as soon as possible.

Reply:
> I did not realize that the scheduling of `build.yml` was intended to catch OS changes in the "release" build, since the workflow is titled "Maven PR Build".

Yeah, the downside of doing more than one thing in an action, which you've remedied now.

> I have been working off the assumption that our nightly build wouldn't use the patch for long, in hopes that it would be fixed upstream "soon enough". For me, the immediate need it satisfies is providing, for the cantaloupe-auth-delegate integration tests, a version of Cantaloupe without (AFAIK) a buggy auth implementation. At the point that we consider the patched image safe to use in production, I would hope we've already submitted the patch in an upstream PR. I would like to stop using the patch in our nightly builds as soon as possible.

Okay.